django/contrib/admin/views/main.py

```diff
@@ -315,10 +315,19 @@ def get_results(self, request):
         # Note this isn't necessarily the same as result_count in the case of
         # no filtering. Filters defined in list_filters may still apply some
         # default filtering which may be removed with query parameters.
-        if self.model_admin.show_full_result_count:
+        if not self.model_admin.show_full_result_count:
+            full_result_count = None
+        elif (
+            self.has_active_filters
+            or isinstance(self.root_queryset, Manager)
+            or self.queryset.query.where
```

Comment:
This could be a check of `self.root_queryset.query.where == self.queryset.query.where`

Reply:
What about other cases where the number of returned rows might changed? e.g. by selecting related columns from multi-valued relations.